### PR TITLE
Support sorting nested anonymous structs

### LINKED ISSUE #1: Support sorting nested anonymous structs
running `gosortstructs` on the following struct has no effect
```go
type MyStruct struct {
	MyField struct {
		B string
		A string
	}
}
```

Ideally it should be sorted as
```go
type MyStruct struct {
	MyField struct {
		A string
		B string
	}
}
```

### PR DESCRIPTION
**Pull request by meblum**
Fixes #1

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -171,45 +171,12 @@ func fieldName(x interface{}) *dst.Ident {
...
-		if len(c.line) != 0 {
-			// convert dst node to ast and get position
-			startLNo := c.fset.Position(c.dec.Ast.Nodes[x].Pos()).Line
-			endLNo := c.fset.Position(c.dec.Ast.Nodes[x].End()).Line
-			if !(startLNo <= c.start && c.end <= endLNo) {
-				return true
-			}
-		}
+	var sortStructFields func(*dst.StructType)
+	sortStructFields = func(s *dst.StructType) {
```
Comment (danishprakash) [inline main.go:176]:
```suggestion
	sortStructFields := func(s *dst.StructType) {
```

unless it goes out scope when you recursively try to use it, shouldn't be the case though

File: main.go [hunk 2]
```diff
@@ -230,34 +196,68 @@ func (c *config) modify(f *dst.File) (*dst.File, error) {
...
+		}
+
+		if c.strct != "" && t.Name.Name == c.strct {
+			foundOne = true
+		}
+
+		s, ok := t.Type.(*dst.StructType)
+		if !ok {
+			return true
+		}
```
Comment (danishprakash) [inline main.go:252]: The code is clearly much more readable now but any reason to get rid of the comments? I found at least some of those helpful.

**Review by danishprakash — COMMENTED** [review]
Looks good, thanks a lot for this. Too bad there are no tests yet which might've made the contribution experience a tad bit easier. I'll add a note to myself for adding tests.

**Comment by meblum** [comment]
Please feel free to modify the pull request as needed, don't think I'll have the time to refactor it within the next while.

File: main.go (hunk 1 above)
Comment (meblum) [reply main.go:176]: we can't do this, we won't be able to call the function on line 228

**Comment by meblum** [comment]
found the time :) forced pushed an update where I reapplied the comments.

**Comment by danishprakash** [comment]
Thanks @meblum!

**Comment by meblum** [comment]
Thanks to you @danishprakash Would you please push a new tag so that go install returns the latest version?

**Comment by danishprakash** [comment]
Done

**Pull request merged**